xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/XDOMBuilder.java

```diff
@@ -52,13 +52,14 @@ public XDOMBuilder()
      */
     public XDOM getXDOM()
     {
-        List<Block> blocks = endBlockList();
-
-        if (!this.stack.isEmpty()) {
+        
+        if (!this.stack.size()==1) {
```

Comment:
This is not exactly the same meaning as previous code. You should check if the size is > 1.